gcc/rust/lint/rust-lint-scan-deadcode.h

```diff
@@ -46,15 +46,34 @@ class ScanDeadcode : public MarkLiveBase
   void visit (HIR::Function &function) override
   {
     HirId hirId = function.get_mappings ().get_hirid ();
-    warning (hirId, function.get_locus (), "function",
-	     function.get_function_name (), "used");
+    if (should_warn (hirId))
+      {
+	warning (function.get_locus (), "function",
```

Comment:
I don't personally like this wrapper around the warning. I would prefer each of the warnings were done by calling rust_warning_at directly.

Reply:
Ok, gotcha